xdsl/dialects/vector.py

```diff
@@ -757,6 +761,286 @@ def _print_attrs(self, printer: Printer):
+            permutation_map=permutation_map,
+            in_bounds=in_bounds,
+            result_type=vector_type,
+        )
+
+
+@irdl_op_definition
+class TransferWriteOp(VectorTransferOperation):
+    name = "vector.transfer_write"
+
+    vector = operand_def(VectorType[Attribute])
+    source = operand_def(TensorOrMemRefOf(Attribute))
+    indices = var_operand_def(IndexType)
+    mask = opt_operand_def(VectorType[I1])
+
+    permutation_map = prop_def(AffineMapAttr)
+
+    result = opt_result_def(TensorType[Attribute])
+
+    irdl_options = [AttrSizedOperandSegments(as_property=True), ParsePropInAttrDict()]
+
+    def __init__(
+        self,
+        vector: SSAValue | Operation,
+        source: SSAValue | Operation,
+        indices: Sequence[SSAValue | Operation],
+        in_bounds: ArrayAttr[BoolAttr],
+        mask: SSAValue | Operation | None = None,
+        permutation_map: AffineMapAttr | None = None,
+        result_type: TensorType[Attribute] | None = None,
+    ):
+        super().__init__(
+            operands=[vector, source, indices, mask],
+            properties={"in_bounds": in_bounds, "permutation_map": permutation_map},
+            result_types=[result_type],
+        )
+
+    def print(self, printer: Printer):
+        printer.print_string(" ", indent=0)
+        printer.print_operand(self.vector)
+        printer.print_string(", ", indent=0)
+        printer.print_operand(self.source)
+        printer.print_string("[", indent=0)
+        printer.print_list(self.indices, printer.print_operand)
+        printer.print_string("]", indent=0)
+        if self.mask is not None:
+            printer.print_string(", ", indent=0)
+            printer.print_ssa_value(self.mask)
+        self._print_attrs(printer)
+        printer.print_string(" : ", indent=0)
+        printer.print_attribute(self.vector.type)
+        printer.print_string(", ", indent=0)
+        printer.print_attribute(self.source.type)
+
+    @classmethod
+    def parse(cls, parser: Parser) -> TransferWriteOp:
+        vector = parser.parse_unresolved_operand()
+        parser.parse_punctuation(",")
+        source = parser.parse_unresolved_operand()
+        indices = parser.parse_comma_separated_list(
+            Parser.Delimiter.SQUARE, parser.parse_operand
+        )
+        if parser.parse_optional_punctuation(","):
+            mask_start_pos = parser.pos
+            mask = parser.parse_unresolved_operand()
+            mask_end_pos = parser.pos
+        else:
+            mask_start_pos = None
+            mask = None
+            mask_end_pos = None
+        attributes_dict = parser.parse_optional_attr_dict()
+
+        types_pos = parser.pos
+        parser.parse_punctuation(":")
+        vector_type = parser.parse_type()
+        parser.parse_punctuation(",")
+        shaped_type = parser.parse_type()
+
+        vector = parser.resolve_operand(vector, vector_type)
+        source = parser.resolve_operand(source, shaped_type)
+
```

Comment:
How much can the next part be shared between the two operations?

Reply:
I factored it out